lib/flight_job/models/script/metadata.rb

```diff
@@ -59,15 +60,10 @@ class Metadata < Metadata::BaseMetadata
         }
       })
 
-      validate do
-        schema_errors = SCHEMA.validate(self.to_hash).to_a
-        unless schema_errors.empty?
-          path_tag = File.exist?(@path) ? @path : @parent.id
-          FlightJob.logger.info("Invalid metadata: #{path_tag}\n")
-          JSONSchemaErrorLogger.new(schema_errors, :info).log
-          errors.add(:metadata, 'is not valid')
-        end
-      end
+      validates_with JsonSchemaValidator,
+                     schema: SCHEMA,
+                     json_method: :to_hash,
+                     error_key: :metadata
```

Comment:
This looks good, but its highlighted that we've lost validation of a scripts metadata.  Can you add the following to `Script` please.

```ruby
    validate do
      unless metadata.valid?
        messages = metadata.errors.map { |e| e.message }
        errors.add(:metadata, messages.join("; "))
      end
    end
```

Reply:
It should then be possible to remove the explicit calls to `script.metadata.valid?` in the specs as those validations should run on the call to `script.valid?(:load)`.